keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Locatable.kt

```diff
@@ -20,11 +20,31 @@ package com.netflix.spinnaker.keel.api
 /**
  * An object which is located in an account and region
  */
-interface Locatable<T : RegionSpec> : ResourceSpec {
-  val locations: Locations<T>
+interface Locatable<T : Locations<*>> : ResourceSpec {
+  val locations: T
 }
 
-data class Locations<T : RegionSpec>(
-  val accountName: String,
+interface Locations<T> {
+  val accountName: String
+  /**
+   * If not specified here, this should be derived from [subnet] or use a default VPC name.
+   */
+  val vpcName: String?
   val regions: Set<T>
-)
+}
+
+data class SubnetAwareLocations(
+  override val accountName: String,
+  override val vpcName: String?,
+  /**
+   * If not specified here, this should be derived from a default subnet purpose using [vpcName].
+   */
+  val subnet: String?,
+  override val regions: Set<RegionSpec>
+) : Locations<RegionSpec>
+
+data class SimpleLocations(
+  override val accountName: String,
+  override val vpcName: String?,
+  override val regions: Set<String>
+) : Locations<String>
```

Comment:
Instead of one type of `Locations` and 2 types of `RegionSpec` there are now 2 types of `Locations`, one of which just has regions as a `Set<String>` (so no more custom Jackson handling).